perun-registrar-lib/src/test/java/cz/metacentrum/perun/registrar/RegistrarBaseIntegrationTest.java

```diff
@@ -658,6 +657,50 @@ public void testEmbeddedGroupsSubmission_groupAutoApprove() throws PerunExceptio
 				.anyMatch(member -> member.getUserId() == user.getId());
 	}
 
+	@Test
+	public void testEmbeddedGroupsSubmission_initEmbeddedConflict() throws PerunException {
+		GroupsManager groupsManager = perun.getGroupsManager();
+		ApplicationForm form = registrarManager.getFormForVo(vo);
+		registrarManager.updateForm(session, form);
+
+		// create embedded groups form item
+		ApplicationFormItem embeddedGroupsItem = new ApplicationFormItem();
+		embeddedGroupsItem.setType(ApplicationFormItem.Type.EMBEDDED_GROUP_APPLICATION);
+		embeddedGroupsItem.setShortname("embeddedGroups");
+		embeddedGroupsItem = registrarManager.addFormItem(session, form, embeddedGroupsItem);
+		registrarManager.updateFormItems(session, form, Collections.singletonList(embeddedGroupsItem));
+
+		// create group in VO, generate group application form
+		Group group1 = new Group("GroupA", "Cool folks");
+		groupsManager.createGroup(session, vo, group1);
+		registrarManager.createApplicationFormInGroup(session, group1);
+		groupsManager.addGroupsToAutoRegistration(session, List.of(group1));
+
+		// create user
+		User user = new User(-1, "Jo", "Doe", "", "", "");
+		user = perun.getUsersManagerBl().createUser(session, user);
+
+		Application application = prepareApplicationToVo(user);
+
+		// set embedded groups in VO application
+		String embGroupsValue = String.format("Group A#%d", group1.getId());
+		List<ApplicationFormItemData> appItemsData = new ArrayList<>();
+		appItemsData.add(new ApplicationFormItemData(embeddedGroupsItem, "Embedded groups", embGroupsValue, "0"));
+		registrarManager.submitApplication(session, application, appItemsData);
+
+		// prepare group application and approve
```

Comment:
This comment makes it a little bit confusing what is actually approved.
```suggestion
		// prepare group application and approve vo application
```